api/client/container/start.go

```diff
@@ -103,14 +103,25 @@ func runStart(dockerCli *client.DockerCli, opts *startOptions) error {
 			return errHijack
 		})
 
-		// 3. Start the container.
+		// 3. We should open a channel for receiving status code of the container
+		// no matter it's detached, removed on daemon side(--rm) or exit normally.
+		statusChan, err := waitExitOrRemoved(dockerCli, context.Background(), c.ID, c.HostConfig.AutoRemove)
+		if err != nil {
+			return fmt.Errorf("Error waiting container's exit code: %v", err)
```

Comment:
We may want to store this error and return it later (instead of waiting on the status chan), or at least improve the error message since it's weird to say that you couldn't wait on a container's exit code before it's even started.